src/v/storage/segment.h

```diff
@@ -220,6 +225,11 @@ class segment {
 
     absl::btree_map<size_t, model::offset> _inflight;
 
+    // Timestamp from server time of first data written to this segment,
+    // field is set when a raft_data batch is appended.
+    // Used to implement segment.ms
+    std::optional<std::chrono::system_clock::time_point> _first_write;
```

Comment:
This could be a ss::lowres_clock -- it's cheaper than using a real system time, and for our purposes a time from a non-system-time steady clock is suitable, since we are only looking at relative duration.